content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -0,0 +1,331 @@
+
+This is an article about Sockets. It will give you a glimpse of how clients and servers communicate over the internet.
+
+Table of content:
+- [What is a socket](#what-is-a-socket)
+- [Programming on Server-side](#programming-on-server-side)
+- [](#)
+- [](#)
+- [](#)
+
+### Introduction
+Socket programming is a way of connecting two computers on a network to allow the exchange of information between the two machines that are running over `JRE`(Java Runtime Environment). The connection can be through **connection-oriented protocol** or **connectionless** but for our case, we will be using connection-oriented which is TCP/IP.
+
+TCP is a connection-oriented protocol, which means that before data can be transferred, two computers must first establish a connection.
+
+Connectionless -For this type of connection, UDP protocol (User Datagram Protocol)also a layer 2 protocol, is used to provide a good picture where you do not necessarily need every packet to arrive at its desired destination.
+
+We will be using Client\Server architecture to illustrate more on sockets. Client and server communicate by writing to and reading from socket connection thus for a connection to happen between the server and a client, a socket must be present. The two machines should know each other's network information.
+
+### What is a Socket?
+A socket is an **endpoint** of communication, acting as a connection between two computers on a network. It contains a socket number that the TCP layer can use to identify the application to which data will be transmitted. An endpoint is made up of a port number and an IP address.
+
+There are two types of sockets
+
+1. Server socket - which waits for a request from clients
+2. Client socket - can be used to send and receive a request from clients.
+
+The client must know two information about the server:
+1.IP address of Server and 
+2.Port number
+
+>Port number 0-1024 are reserved for administrative purposes(e,g 21 for FTP,23 for Telnet,25 for email and 80 for HTTP).
+
+This happens when a TCP connection is being used between two machines:-
```

Comment:
>This happens when a TCP connection is being used between two machines

I would probably reword this as something like: "Whenever we create a socket connection in Java, the following steps would take place:" I just feel like what you had sounded a little awkward to me.